### PR TITLE
feat(workflow-state): assign to content-item and update workflow-state support

### PR DESCRIPTION
**Pull request by easen-amp**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Tests (`npm run test`) for the changes have been added (for bug fixes / features)
- [x] Docs (`npm run doc`) have been reviewed and added / updated if needed (for bug fixes / features)
- [x] PR title and git commit messages conform to the [conventionalcommits](https://www.conventionalcommits.org/en/v1.0.0/) specification

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behaviour?

Missing workflow states

## What is the new behaviour?

- get, list & update workflow states
- assign workflow states to a content item

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by simonmudd — APPROVED** [review]

File: src/lib/model/WorkflowState.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  /**
+   * Resources and actions related to a WorkflowState
+   */
+  public readonly related = {
+    /**
+     * Retrieves the Hub this webhook is stored in
+     */
+    hub: (): Promise<Hub> => this.fetchLinkedResource('hub', {}, Hub),
+
+    update: (mutation: WorkflowState): Promise<WorkflowState> =>
```
Comment (rs-amp) [inline src/lib/model/WorkflowState.ts:56]: No documentation here?

**Review by rs-amp — CHANGES_REQUESTED** [review]
Just one minor doc omission. Looks good otherwise.

**Review by Benj0s — APPROVED** [review]

File: src/lib/model/WorkflowState.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  /**
+   * Timestamp representing when the Workflow State was last updated in ISO 8601 format
+   */
+  public lastModifiedDate?: string;
+
+  /**
+   * Assigned colour in the format of `rgb(R,B,G)`
+   */
+  public 'color': string;
```
Comment (rs-amp) [inline src/lib/model/WorkflowState.ts:46]: Missing newline after this

Comment (rs-amp) [reply src/lib/model/WorkflowState.ts:46]: Also, is (R,B,G) really the order?

**Review by rs-amp — CHANGES_REQUESTED** [review]

**Review by c-bibby — APPROVED** [review]

**Review by rs-amp — APPROVED** [review]

**Pull request merged**